source/FreeRTOS_IP_Utils.c

```diff
@@ -220,6 +224,10 @@ static NetworkBufferDescriptor_t * prvPacketBuffer_to_NetworkBuffer( const void
     else
     {
         /* Obtain the network buffer from the zero copy pointer. */
+
+        /* copy pointer value to uintptr_t which is guaranteed to fit in a
+         * pointer */
+        /* coverity[misra_c_2012_rule_11_6_violation] */
         uxBuffer = ipPOINTER_CAST( uintptr_t, pvBuffer );
 
```

Comment:
@alfred2g, thanks for this PR.

Please consider the following change:
~~~diff
-uxBuffer = ipPOINTER_CAST( uintptr_t, pvBuffer );
+uxBuffer = ( uintptr_t ) pvBuffer;
~~~
I once introduced the macro `ipPOINTER_CAST()` in order to silence warnings about casts. Let's stop using that.

In a later version we used a (complex) set of pseudo functions that did the same, like e.g. `ipCAST_CONST_PTR_TO_CONST_TYPE_PTR`. Recently, Aniruddha removed these, and (fortunately) we're back to the original code. Along with some messages for coverity.

Reply:
I agree with Hein. Let us remove that (`ipPOINTER_CAST`) macro.